pkg/solana/chain.go

```diff
@@ -574,12 +574,13 @@ func (c *chain) sendTx(ctx context.Context, from, to string, amount *big.Int, ba
 	}
 
 	chainTxm := c.TxManager()
-	err = chainTxm.Enqueue(ctx, "", tx, nil,
+	err = chainTxm.Enqueue(ctx, "", tx, nil, blockhash.Value.LastValidBlockHeight,
```

Comment:
I think we can avoid changing the `Enqueue` method signature here. Not an issue if we want to though. I think we can add the `c.reader.LatestBlockhash(ctx)` logic in `sendWithRetry` and just overwrite `tx.Message.RecentBlockhash`. That way the TXM can own assigning the block hashes to transactions.